test/rubygems/test_gem.rb

```diff
@@ -29,6 +29,17 @@ def setup
     util_remove_interrupt_command
   end
 
+  def teardown
+    super
+
+    change_version @version if defined? @version
+  end
+
+  def change_version new_version
+    @version = Gem.instance_variable_get :@rubygems_version
```

Comment:
this conflicts with `test/rubygems/test_gem_source_git.rb:10`, maybe use `@gem_orig_version ||=`